packages/blobdex/src/index.js

```diff
@@ -0,0 +1,81 @@
+import { base58btc } from 'multiformats/bases/base58'
+export * from './types.js'
+
+export class Blobdex {
+  #items
+
+  constructor() {
+    this.#items = new Map()
+  }
+
+  /**
+   * len numbers the count of items in this Blobdex.
+   *
+   * @returns {number}
+   */
+  len() {
+    return this.#items.size
+  }
+
+  /**
+   * add adds a new item to this Blobdex multihash digest bytes key.
+   *
+   * @param {import('multiformats').MultihashDigest} mh
+   * @param {number} offset
+   * @param {number} length
+   */
+  add(mh, offset, length) {
+    const b58str = base58btc.encode(mh.bytes)
+    this.addStr(b58str, offset, length)
```

Comment:
Perhaps validate `offset` and `length` are integers and > 0? `Number.isInteger(offset) && offset > 0`